src/reason_parser.mly

```diff
@@ -1216,65 +1217,120 @@ parse_pattern:
 
 mark_position_mod(X):
 	x = X
-	{ {x with pmod_loc = {x.pmod_loc with loc_start = $symbolstartpos; loc_end = $endpos}} }
+	{
+    if x.pmod_loc.loc_start.Lexing.pos_lnum <> 0 then
```

Comment:
The repositioning of the attributes has consequences for the start_pos of many first tokens. Instead of always overwriting it we now only do it when there is none.